November 19 2024 - Add Vincent Latombe spotlight (#219)
jenkins-infra/contributor-spotlight: src/pages/contributors/vincent-latombe.adoc

= Vincent Latombe
:page-name: Vincent Latombe
:page-linkedin:
:page-twitter: 
:page-github: vlatombe
:page-email:
:page-image: avatar/vincent-latombe.jpg
:page-pronouns: He/Him/His
:page-location: Villefranche sur Saone, France
:page-firstcommit: 2011
:page-datepublished: 2024-11-19
:page-featured: true

> :page-intro: Vincent Latombe is a software engineer and long-standing Jenkins contributor who has worked on many areas since his first introduction. From plugins, to Jenkins core, to test parallelization, Vincent has been part of several improvements and new features over the last decade. He has held an interest in computers and technology since childhood, fostered by his parents encouragement. They like to say that Vincent broke his first keyboard at 18 months old, but Vincent maintains that he's never broken a keyboard in his life. His interest in programming languages started with some Commodore practice, but there was a lack of use cases to apply what he was learning. Vincent graduated from school and found his footing through roles in engineering and developer relations, where he eventually got connected with other Jenkins contributors and even Kohsuke Kawaguchi himself. Vincent has worn many hats when it comes to Jenkins and continues to find new ways to contribute to and enhance the project. When he's not working on HA, Vincent likes to balance his tech work with getting outside and being active in different ways. He's played badminton for half his life and enjoys biking around with his two children. He's also been passing on his love of computers to his children, hoping to foster the same sense of creativity that he still possesses.

Author: I definitely agree that it would be better, but unfortunately breaking it up line by line will not render on the overview page as we intend. The intended result is to have the text create an indication for the user to read further, by forcing an ellipses to be shown in the featured spotlight box:
<img width="702" alt="Screenshot 2024-11-08 at 2 44 19 PM" src="https://github.com/user-attachments/assets/1def0ef1-f32e-40c1-9ae0-0b9c6892cda7">

I've adjusted it to be a bit more separated and not one giant paragraph without losing the `...`

Vincent graduated from school and found his footing through roles in engineering and developer relations, where he eventually got connected with other Jenkins contributors and even Kohsuke Kawaguchi himself. Vincent has worn many hats when it comes to Jenkins and continues to find new ways to contribute to and enhance the project. When he's not working on HA, Vincent likes to balance his tech work with getting outside and being active in different ways. He's played badminton for half his life and enjoys biking around with his two children. He's also been passing on his love of computers to his children, hoping to foster the same sense of creativity that he still possesses.

== What is your background prior to contributing to Jenkins?

> I graduated high school in 2001 and then, as is customary in France, attended preparatory school in order to go to engineering school.
> Only the students from the preparatory school with the best grades are selected to take the written and oral examinations for engineering school.
> Alternatively, some people decide to attend university, work towards being within the top of their class, and enter the engineering school that way.
> Unfortunately, I missed the top spots so I had to go for a third year.
> After finally achieving the marks I needed, I went to an engineering school in Grenoble and studied software engineering.

Author: ```suggestion
I graduated high school in 2001, followed by preparatory school in Paris.
After receiving the marks needed, I went to Ensimag, an engineering school in Grenoble, where I studied telecommunications and software engineering.
```
Would this also work @Vlatombe?

For my final year, I studied in Sweden for eight months, then returned and completed an internship for my masters degree.
After getting my degree, I was hired by Amadeus and worked for them for eight years.

> Initially, I worked on the product, but I quickly switched to the backend and tooling.

Author: ```suggestion
Initially, I worked in a UI team that was embracing the beginning of JS frameworks (Yahoo UI), but after 2 years I switched to a transversal team dealing with tooling and builds.
I was very fond of tooling and figuring out ways to make people more productive.
```

In 2009 or 2010, during the early stages of CI implementation, the company had some Cruise Control instances deployed.
Then we heard about a product called Hudson that was open source. 
It had a nice web interface where you could edit your projects using the UI directly.
Cruise control was driven from XML files as well, but every time you wanted to edit anything, you needed to restart the instance.
It was nowhere as flexible as Hudson.
When we started with Hudson, I started by picking one server, deploying a project with a branch, configuring limited availability, and seeing what we could do with it quickly.

> The SCM we were using at the time was Clear Case from IBM, which was designed to work within an IBM environment.

Author: thanks for clarifying that!

> However, there was a Clear Case plugin maintained by Andrew Bayer.

Author: ```suggestion
At the time, Hudson had a ClearCase plugin maintained by Andrew Bayer.
```

When I started using Hudson, I began discovering some problems with my environment.

> I started to contribute fixes, and then after a while, we grew through a deployment.
> We started with building just the main branch, then we started considering more projects, finally leading up to building a development branch.
> My development with Jenkins grew organically over the years and that's how I got involved with the Jenkins project.
> Eventually, Andrew changed companies and left Clear Case, which meant that the Clear Case plugin was no longer being maintained.

Author: ```suggestion
I started to contribute fixes, and then after a while, we grew through a deployment.
We started with building just the main branch of one project, then multiple projects, then all projects.
This was a game changer, especially in an environment where integration phases were a real challenge and people delivering features as quick as they could.
These would result in conflicts that were taking days, if not weeks to resolve, but getting a point of reference and tracking stabilization efforts were key.
Getting high performance hardware was also a challenge back then, since developers would take hours just to compile projects locally.
CPUs were 2 cores, magnetic disks, with no efficient way to parallelize.
My development with Jenkins grew organically over the years and that's how I got involved with the Jenkins project.
Eventually, Andrew changed companies and stopped maintaining ClearCase (to start using Git), leaving the ClearCase plugin up for adoption.
```

While in the developers advocacy role, one of the goals at the time was to reorganize the many product silos.
I was focused on meeting everyone and understanding how each of them was working and why they were working that way.
Then we would try to come up with a unified approach that would work for most people.
Eventually, they reorganized everything and put a transversal team at the top so that they could build some common practices across the whole organization, instead of having different styles.
When I left Amadeus, I think we were up to maybe a hundred executors.
We had around a dozen big machines and each machine had ten executors since we were not yet using virtualization.
It was a pretty big instance with lots of builds and around 600 people using it.

Around 2012 or 2013 I started attending some Jenkins community events and went to FOSDEM.
That's when I met Kohsuke for the first time and other Jenkins contributors as well.
We were in the same situation; contributing to Jenkins from our separate companies.
Later on, when CloudBees did some fundraising, they started to hire some of the community members.

> Kohsuke offered me a position at CloudBees, which I accepted.

Author: ```suggestion
This is when Kohsuke reached out to me to offer a position at Cloudbees, which I eventually accepted.
```

So the best structure was to create one job per branch.
However, it messy and complicated because there was no way to capture what was common between the different jobs.
Obviously, at the time we did not have Pipeline.
Therefore, we chose to build an automation using a template that would copy it over, customize the branch name, and then repeat that for the dozens of projects that were following the same lifecycle but managed as different projects.
All of these projects needed to be managed and we needed to have a way to determine whether a component actually changed, otherwise a build that comes from the main branch would be reused.
Every time we had a big project, it became very costly if we didn't do it in a smart way.
So we always try to optimize for the infrastructure and time because otherwise we pay for the computing processes, storage, network, and other aspects.
Jenkins helped with this optimization and provided a way for us to manage all the jobs that were being run.

== What have you worked on in Jenkins?

Initially, when I got recruited at CloudBees, they were building proprietary plugins on top of Jenkins as well as contributing to the community.

> It was the very beginning of cloud support, which eventually became CI modern.
> It was basically a prototype, where we were trying to generalize how to provision Jenkins instances automatically without things like Cask, Docker, and a lot of other things that were new at the time.

Author: ```suggestion
When I came in, I started to work on a prototype of a deployment of Jenkins in the cloud.
We were trying to generalize how some companies were doing massive deployments of Jenkins on clustered systems like Mesos, but packaged in a product that could be run by any of our customers.
At the time, we started using Mesos and Marathon, while using Docker as a container engine.
Since we were running on AWS (which was still charging by the hour), and Docker volumes did not exist, we had to create our own system to allocate ebs volumes on the fly.
```

== What is your background prior to contributing to Jenkins?

I graduated high school in 2001, followed by preparatory school in Paris.
After receiving the marks needed, I went to Ensimag, an engineering school in Grenoble, where I studied telecommunications and software engineering.
For my final year, I studied in Sweden for eight months, then returned and completed an internship for my masters degree.
After getting my degree, I was hired by Amadeus and worked for them for eight years.

> Initially, I worked in a UI team that was embracing the beginning of JS frameworks (Yahoo UI), but after 2 years I switched to a transversal team dealing with tooling and builds.
> I was very fond of tooling and figuring out ways to make people more productive.

Reviewer: ```suggestion
Initially, I worked in a UI team that was embracing the beginning of JS frameworks (Yahoo UI), but after two years, I switched to a transversal team that dealt with tooling and builds.
I was very fond of tooling and figuring out ways to make people more productive.
```

In 2010, during the early stages of CI implementation, the company had some Cruise Control instances deployed.
Then we heard about a product called Hudson that was open source. 
It had a nice web interface where you could edit your projects using the UI directly.
Cruise control was driven from XML files as well, but every time you wanted to edit anything, you needed to restart the instance.
It was nowhere as flexible as Hudson.
When we started with Hudson, I started by picking one server, deploying a project with a branch and seeing what we could do with it quickly.
The SCM we were using at the time was ClearCase from IBM, which was designed to work within an IBM environment.
At the time, Hudson had a ClearCase plugin maintained by Andrew Bayer.
When I started using Hudson, I began discovering some problems with my environment.

I started to contribute fixes, and then after a while, we grew through a deployment.

> We started with building just the main branch of one project, then multiple projects, then all projects.
> This was a game changer, especially in an environment where integration phases were a real challenge and people delivering features as quick as they could.
> These would result in conflicts that were taking days, if not weeks to resolve, but getting a point of reference and tracking stabilization efforts were key.
> Getting high performance hardware was also a challenge back then, since developers would take hours just to compile projects locally.
> CPUs were 2 cores, magnetic disks, with no efficient way to parallelize.

Reviewer: ```suggestion
We started with building just the main branch of one project, then multiple projects, then all projects.
This was a game changer, especially in an environment where integration phases were a real challenge and people delivering features as quickly as they could.
These would result in conflicts that were taking days, if not weeks to resolve, but getting a point of reference and tracking stabilization efforts were key.
Getting high-performance hardware was also a challenge back then since developers would take hours just to compile projects locally.
CPUs were two cores, and magnetic disks, with no efficient way to parallelize.
```

== What have you worked on in Jenkins?

Initially, when I got recruited at CloudBees, they were building proprietary plugins on top of Jenkins as well as contributing to the community.
When I came in, I started to work on a prototype of a deployment of Jenkins in the cloud.
We were trying to generalize how some companies were doing massive deployments of Jenkins on clustered systems like Mesos, but packaged in a product that could be run by any of our customers.
At the time, we started using Mesos and Marathon, while using Docker as a container engine.
Since we were running on AWS (which was still charging by the hour), and Docker volumes did not exist, we had to create our own system to allocate ebs volumes on the fly.

To focus on my current Jenkins efforts, I've been working on HA support.
The goal is making a Jenkins installation able to span over multiple machines and horizontally scale.

> This has involved a lot of work, because Jenkins historically had a lot of design principles that were counter to scaling that way; a heavy reliance on a file system, holding most of the state in memory.
> We made a lot of patches against Jenkins core in that context.
> In most cases, this involved creating new extension points, to allow users to swap the default implementation for another one that can work in a HA-context.

Reviewer: ```suggestion
This has involved a lot of work because Jenkins historically had a lot of design principles that were counter to scaling that way; a heavy reliance on a file system, holding most of the state in memory.
We made a lot of patches against Jenkins core in that context.
In most cases, this involved creating new extension points, to allow users to swap the default implementation for another one that can work in a HA-context.
```
